sync-core/src/main/java/com/cloudant/sync/sqlite/SQLQueueCallable.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2015 IBM Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+package com.cloudant.sync.sqlite;
+
+import java.util.concurrent.Callable;
+
+/**
+ * A task which performs actions on a SQLDatabase passed in when
+ * called.
+ */
+public abstract class SQLQueueCallable<T> implements Callable<T> {
+
+    private SQLDatabase db;
+    private boolean inTransaction = false;
+
+    @Override
+    final public T call() throws Exception {
+        if(inTransaction){
+            try {
+                db.beginTransaction();
+                T returned =  call(db);
+                db.setTransactionSuccessful();
```

Comment:
I think I'd add a short comment here that `call() throws if transaction should be aborted` so no one makes the mistake I made when originally reading that there's no way to fail a transaction.